pvlib/horizon.py

```diff
@@ -0,0 +1,459 @@
+                view_distance=500, az_len=36):
+    r"""Finds the horizon at point on a dem in pixel coordinates dem_pixel
+
+    Parameters
+    ----------
+    dem_pixel : tuple (int,int)
+        Point on the DEM expressed in pixel coordinates
+    elevation : np.array
+        nxn DEM of elevation values
+    dem_res : float
+        Resolution of the DEM. The default is SRTM 30m
+    view_distance : int
+        Radius of the area of consideration.
+    az_len : int
+        Number of samples on the perimeter of the circle.
+
+    Returns
+    -------
+    azimuth : np.array
+        Numpy array of shape (az_len). Linearly spaced from [0,360]
+    elevation_angles: np.array
+        The calculated elevation values at each point of azimuth
+    profile:
+        The largest elevation in meters on the line between the observer
+        and the highest point on the horizon within view_distance
+
+    """
+    from skimage.draw import line
+    azimuth = np.linspace(0, 360, az_len)
+
+    # Use Midpoint Circle Algo/ Bresenham's circle
+    pts = _bresenham_circ(view_distance, dem_pixel)
+
+    # sort circle points and resample to desired number of points (az_len)
+    pts = _sort_circ(pts, center=dem_pixel, az_len=az_len)
+    x0, y0 = dem_pixel
+    profile = np.zeros(azimuth.shape)
+    elevation_angles = np.zeros(azimuth.shape)
+    x_bound, y_bound = elevation.shape
+    for az in range(az_len):
+        source_lon = pts[1][az]
+        source_lat = pts[0][az]
+        # check for outmapping
+        # for now, if out of the DEM, assign to the border
+        if source_lon >= y_bound:
+            source_lon = y_bound-1
+        elif source_lon <= 0:
+            source_lon = 0
+        if source_lat >= y_bound:
+            source_lat = y_bound-1
+        elif source_lat <= 0:
+            source_lat = 0
+
+        target_lon = int(y0)
+        target_lat = int(x0)
+
+        # draw a line from observer to horizon and record points
+        rr, cc = line(source_lon, source_lat, target_lon, target_lat)
+
+        # get the highest elevation on the line
+        elvs_on_line = elevation[rr, cc]
+        idx = np.stack((rr, cc), axis=-1)
+        highest_elv = np.max(elvs_on_line)
+        highest_point = idx[np.argmax(elvs_on_line)]
+        high_x, high_y = tuple(highest_point)
+
+        # convert from altitude in m to elevation degrees.
+        xdist = np.abs(highest_point[0]-x0)
+        ydist = highest_elv - elevation[y0][x0]
+        elv_ang = np.arctan(ydist/xdist)
+        elevation_angles[az] = np.rad2deg(elv_ang)
+        profile[az] = highest_elv
+    return azimuth, elevation_angles, profile
+
+
+def calculate_dtf(horizon_azimuths, horizon_angles,
+                  surface_tilt, surface_azimuth):
+    r"""Author: JPalakapillyKWH
+    Calculate the diffuse tilt factor for a tilted plane that is adjusted
+    with for horizon profile. The idea for a diffuse tilt factor is explained
+    in [1].
+    Parameters
```

Comment:
Include a `Notes` section to credit the author JPalakapillyKWH

```suggestion
    r"""Calculate the diffuse tilt factor for a tilted plane that is adjusted
    with for horizon profile.

    The idea for a diffuse tilt factor is explained
    in [1].

    Parameters
```